src/extensions/default/Inline3DParametersEditor/Inline3DParameterEditor.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * File based on InlineColorEditor/InlineColorEditor.js
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var InlineWidget            = brackets.getModule("editor/InlineWidget").InlineWidget,
+        ParameterEditor         = require("ParameterEditor").ParameterEditor,
+        Inline3dParametersUtils = require("Parameters3DUtils");
+
+    /** @type {number} Global var used to provide a unique ID for each parameter editor instance's _origin field. */
+    var lastOriginId = 1;
+
+    /**
+     * Inline widget containing a ParameterEditor control
+     * @param {!string} parameters  Initial Parameters
+     * @param {!CodeMirror.TextMarker} marker
+     */
+    function Inline3DParameterEditor(parameters, marker, tag) {
+        this._parameters = parameters;
+        this._marker = marker;
+        this._tag = tag;
+        this._isOwnChange = false;
+        this._isHostChange = false;
+        this._origin = "+InlineParameterditor_" + (lastOriginId++);
+
+        this._handleParametersChange = this._handleParametersChange.bind(this);
+        this._handleHostDocumentChange = this._handleHostDocumentChange.bind(this);
+        InlineWidget.call(this);
```

Comment:
Move this up to the first thing you do in this constructor, so that in case any of these other properties are also added by the base constructor, you properly overwrite.